core/modules/path/path.admin.inc

```diff
@@ -643,9 +643,14 @@ function path_bulk_update_form($form, &$form_state) {
       // vocabularies.
       if (isset($path_info['pattern items'])) {
         foreach ($path_info['pattern items'] as $item => $item_label) {
+          $human_name = $item;
+          // Use the human name in the row title for vocabularies, not the machine name they are keyed by.
+          if ($type == 'taxonomy_term') {
+            $human_name = substr(strip_tags($item_label), 26, -5);
```

Comment:
your intuition is correct; what we need here is to expose the Human Name as part of the `pattern items` array.

The `$path_info` variable is set via line 624:

```php
  $all_path_info = path_get_info();
```

That function in turn calls all the callbacks from any module that implements that hook; in this case the `taxonomy` module; The hook implementation in `taxonomy` is in the file: `taxonomy.path.inc` on line 10.  In the for loop that is populating the `$info` variable we need something like this in both branches of the `if`:

```php
$info['taxonomy_term']['pattern items'][$vocabulary_name . '-human'] = $vocabulary->name;
```

That will expose the human name here so you can just grab it instead of having to to the string manipulations here.

Reply:
...or just do this:

```suggestion
            $item = taxonomy_vocabulary_machine_name_load($item)->name;
```

😉

Reply:
@klonos @isabelhaas  the issue here is performance; this submit handler is already passing off to any module that implements `hook_path_info` and in `taxonomy_path_info()` we are already loading the vocabularies in `$vocabularies = taxonomy_get_vocabularies();` so loading it again at this step is more memory and less efficient.

It would work but is not optimized.

Reply:
@isabelhaas I also see that putting the human name here:

```php
$info['taxonomy_term']['pattern items'][$vocabulary_name . '-human'] = $vocabulary->name;
```

makes too many checkboxes; :) ; we can put it here instead:

```php
$info['taxonomy_term'][$vocabulary_name . '-human'] = $vocabulary->name;
```

Reply:
@serundeputy Thanks for the help! I did what you suggested, could you take a look and see if what's there now looks correct?